rotkehlchen/chain/evm/contracts.py

```diff
@@ -131,6 +135,115 @@ def decode_event(
         return decode_event_data_abi(tx_log=tx_log, event_abi=event_abi)  # type: ignore
 
 
+def try_download_abi_from_remote(name: str) -> Optional[list[dict[str, Any]]]:
+    """
+    Checks whether abi with the given name exists on remote.
+    If it does, downloads it and adds to globaldb.
+
+    Returns the abi if found and None otherwise.
+    """
+    abi_url = f'{BASE_CONTRACT_DATA_URL}/abi/{name}.json'
+    try:
+        abi_data = request_get(abi_url)
+    except (RemoteError, UnableToDecryptRemoteData) as e:
+        log.error(f'Could not find abi {name} in rotki data repo due to: {str(e)}')
+        return None
+
+    if isinstance(abi_data, list) is False:
+        log.error(f'Abi should be a list. But we got {abi_data}')
+        return None
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        write_cursor.execute(
+            'INSERT INTO contract_abi(name, value) VALUES(?, ?)',
+            (name, json.dumps(abi_data, separators=(',', ':'))),
+        )
+
+    return abi_data  # type: ignore[return-value]  # abi_data is a list here for sure
```

Comment:
I'm completely against this ignore types. Can you maybe overload it or in `request_get_dict` we use assertion, imo that is enough here

Reply:
I actually made an `isinstance` check that it is a list a few lines above. I am confused why mypy thinks that the type here is not a list only. I'll try to solve it.